### ISSUE TITLE
using xrdp v0.9.9.1 with Guacamole and drive redirection problem only w/downloads not uploads

### ISSUE THREAD

**Issue by bmullan** [neutrinolabs/xrdp#1382]
I've been using xrdp with a guacamole based project I have worked on for several years.

https://github.com/bmullan/ciab-remote-desktop

TLDR - Host is an Ubuntu 18.04 (cloud/vm or physical server).   Guacamole, tomcat, mysql, nginx run in an LXD container called ciab-guac on the server

User remote desktops (Mate) run in a 2nd LXD container (CN1) on the same server.

Both Containers are also Ubuntu 18.04 also running MATE Desktop Environment.

Nothing runs in the Host Server but LXD everything else is in the 2 LXD containers (**ciab-guac and CN1**)

Everything works great (sound, printing) except I have one problem with File Redirection and that is only in 1 direction ... uploads work file, downloads don't work.

I am using **XRDP v0.9.9.1** (0.9.10 is available I know).

I edited /etc/xrdp/sesman.ini to set the shared drive name to "shared-drive".

In Guacamole I configure the CN1 container's "connection" to:
Enable Drive   checked
Drive Name:   shared-drive
Drive Path:     the logged in User's Desktop folder
and I set guacamole config to use rdpdr

After doing all this config... file uploads work fine.  I just drag & drop a file from my local desktop onto the remote desktop and it gets put into the user's "shared-drive/GUACFS" folder.

However, for downloads from the remote Desktop if I copy a file or drag/drop it into the user's "shared-drive/GUAFS/download" folder the file is never transferred to my local machine?

Any suggestions how I can troubleshoot this from the xrdp side of things?

Also... 
I did see that in XRDP v0.9.10 there were 2 more "fixes" for drive redirection but I am not sure if this might be fixed by those or not after reading about them:

**

> Fix the issue reconnecting to session causes duplicate drive entries in fuse fs **#1299**
> Improve remote file lookup in drive redirection **#996 #1327**

**

Thanks for any ideas.
Brian

**Comment by bmullan** [comment]
think I fixed this via change in guacamole config.. sorry for the noise

**Comment by pingou2712** [comment]
Hi,

I have the same problem... Long time and no answer :( How fix it???? plzzzzzzz
(Sorry for my english, i'm french :P )

Thanks for the help,
Vincent

**Comment by bmullan** [comment]
I think v0.9.10 fixed that
At least it works on my CIAB Remote Desktop System project

https://github.com/bmullan/ciab-remote-desktop

**Comment by bmullan** [comment]
FYI

on ubuntu 18.04 I used Thiago Marten's PPA to install xrdp v0.9.10

https://launchpad.net/~martinx/+archive/ubuntu/xrdp-next?field.series_filter=bionic

**Comment by pingou2712** [comment]
ok..... it's a new machine and my old machine works perfectly... i don't understand why it's not work here.... :( i test tomorrow new version of xrdp....
Thank you for help

**Comment by pingou2712** [comment]
Last version of xrdp tested (0.9.11)... same problem....
error:
guacd[686]: File open refused (-2): "/Download/test"
